libibumad/umad.h

```diff
@@ -54,6 +54,21 @@ typedef __be16 __attribute__((deprecated)) be16_t;
 typedef __be32 __attribute__((deprecated)) be32_t;
 typedef __be64 __attribute__((deprecated)) be64_t;
 
+typedef union _ib_gid
+{
+	uint8_t	raw[16];
+
+	struct {
+		__be64	prefix;
+		__be64	interface_id;
+	} unicast;
+
+	struct {
+		uint8_t	header[2];
+		uint8_t	raw_group_id[14];
+	} multicast;
+} ib_gid_t;
```

Comment:
What would OpenSM configure test do ? Check for (some member in) ib_gid_t in umad.h (via AC_CHECK_MEMBER ?) and if present use that rather than the one in ib_types.h ?

This is going to require release of OpenSM to go with rdma-core which supports this.

It may also affect other tools which include ib_types.h (infiniband-diags).